libs/utils/android/workloads/vellamo.py

```diff
@@ -0,0 +1,175 @@
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from android import Workload
+from time import sleep
+
+import os
+import collections
+import copy
+import logging
+
+VELLAMO_DB_PATH = '/data/data/com.quicinc.vellamo/files'
+VELLAMO_SCORE_NAME = 'chapterscores.json'
+VELLAMO_TESTS = ['BROWSER', 'METAL', 'MULTI']
+
+class Vellamo(Workload):
+    """
+    Android Vellamo workload
+    """
+
+    # Package required by this workload
+    package = 'com.quicinc.vellamo'
+    activity = 'com.quicinc.vellamo.main.MainActivity'
+
+    # Setup logger
+    logger = logging.getLogger('Vellamo')
+    logger.setLevel(logging.DEBUG)
+
+    def __init__(self, test_env):
+        super(Vellamo, self).__init__(test_env)
+        logging.debug('%14s - Workload created', 'Vellamo')
+
+    def wload_run(self, exp_dir, test_name, collect='', emeter_channel=None):
+        """
+        Run single Vellamo workload. Returns a collection of results.
+
+        :param exp_dir: Path to experiment directory where to store results.
+        :type exp_dir: str
+
+        :param test_name: Name of the test to run
+        :type test_name: str
+
+        :param collect: Specifies what to collect. Possible values:
+            - 'energy'
+        :type collect: list(str)
+
+        :param emeter_channel: Energy meter channel from which data is
+            collected
+        :type emeter_channel: str
+        """
+
+        if test_name.upper() not in VELLAMO_TESTS:
+            raise ValueError('Vellamo workload [%s] not supported', test_name)
+
+        # Set parameter depending on test
+        self.logger.debug('Start Vellamo Benchmark [%s]', test_name)
+        test_x, test_y = (0, 0)
+        sleep_time = 0
+        if test_name.upper() == 'BROWSER':
+            test_x, test_y = (970, 740)
+            sleep_time = 3.5
+        elif test_name.upper() == 'METAL':
+            test_x, test_y  = (970, 1275)
+            sleep_time = 1
+        elif test_name.upper() == 'MULTI':
+            test_x, test_y = (970, 1555)
+            sleep_time = 3.5
+
+        nrg_data, nrg_file = None, None
+
+        # Unlock device screen (assume no password required)
+        self.target.execute('input keyevent 82')
+
+        # Set max brightness
+        Screen.set_brightness(self.target, auto=False, percent=100)
+
+        self.target.execute('pm clear {}'.format(self.package))
+        System.start(self.target, self.package, self.activity)
+        # Wait few seconds to make sure the app is loaded
+        sleep(1)
+
+        # Accept EULA
+        System.tap(self.target, 864, 1646)
```

Comment:
This tap call should really be extended to use (by default) a screen percentage instead of these hardcoded pixels values. Can we try to extend this call using the display info returned by devlib?

I would really love to call something like:

```python
# percentage
System.tap(self.target, 50, 80)
#... or absolute:
System.tap(self.target, 864, 1646, absolute=True)
```